File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        merged = [intervals[0]]
        for interval in intervals[1:]:
            if merged[-1][1] >= interval[0]:
                merged[-1][1] = max(interval[1], merged[-1][1])
            else:
                merged.append(interval)
        return merged

File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        
        vector<vector<int>> merged;
        merged.push_back(intervals[0]);
        for(auto &interval: intervals){
            if(merged[merged.size() - 1][1] >= interval[0]){
                merged[merged.size() - 1][1] = max(merged[merged.size() - 1][1], interval[1]);
            }else{
                merged.push_back(interval);
            }
        }
        return merged;
    }
};